app/main/api/handlers/getWallets.ts

```diff
@@ -0,0 +1,15 @@
+import { Contexts, asRuntimeType } from "app/common/runtimeTypes"
+import { Wallets } from "app/common/runtimeTypes/storage/wallets"
+import { SubSystems } from "app/main/system"
+
+/**
+ * Handler function for "getWallets" methods.
+ * It retrieves and returns a list of wallet info objects taken from app state.
+ * @param system
+ * @param params
+ */
+export default async function getWallets(system: SubSystems, params: any) {
+  const _wallets = system.appStateManager.get("wallets")
+
+  return asRuntimeType(_wallets, Wallets, Contexts.STATE)
```

Comment:
I don't understand, this is the picture I have in mind regarding the decoding/encoding of complex types inside handlers:

```
              +---------------------------------------+
    request   |  +-------------------+                |
  ------------+-->  asRuntimeType    |                |
              |  +-------|-----------+                |
              |  +-------v-----------+                |
              |  | logic that works  |                |
              |  | with complex      |                |
              |  | business type     |                |
              |  +-------|-----------+                |
    response  |  +-------v-----------+                |
  <-----------+--+  asObject         |                |
              |  +-------------------+                |
              |                                       |
              |                          a handler    |
              |                                       |
              +---------------------------------------+
```

for me, `system.appStateManager.get("wallets")` should always return the business type we use to represent wallets, `asRuntimeType` should only be used to decode foreign data which in this case is `params`, and the handler should return `asObject` since that guarantees the response is json serializable

Reply:
Yay, you are right, this assertion shouldn't be needed.

This is a leftover from a time when `AppStateManager._state` was private and had no getter.

As it now has a getter, we can directly do this in a typesafe manner:
```ts
return asObject(system.appStateManager.state.wallets, Wallets)
```

Let me patch it.